packages/sdk/src/index.ts

```diff
@@ -672,12 +672,14 @@ class NameGuard {
    *
    * @param {string} contract_address Contract address for the NFT contract (ERC721 and ERC1155 supported).
    * @param {string} token_id The ID of the token (in hex or decimal format).
+   * @param {string} fields Fields with values which will be investigated (e.g. title, collection name, metadata) whether they look like fake .eth ENS name. `title` key is mandatory, for ENS contracts it should be the ENS name.
    * @param {FakeEthNameOptions} options The options for the fake .eth ens name check.
    * @returns {Promise<FakeEthNameCheckResult>}  A promise that resolves with a `FakeEthNameCheckResult`.
    */
   public fakeEthNameCheck(
     contract_address: string,
     token_id: string,
+    fields: Record<string, string>,
```

Comment:
We should probably check that `fields` was present with at least one key.

```ts
if (Object.keys(fields).length === 0) {
  throw new Error("The 'fields' object must contain at least one field.");
}
```

Reply:
It must contain `title` field.

Reply:
Good spot! Let's do two things:

1. Update `Record<string, string>` to be:

```ts
interface FieldsWithRequiredTitle extends Record<string, string> {
    title: string;
}

// ...

fields: FieldsWithRequiredTitle
```

2. Update the validation to be:

```ts
if (!fields.title || typeof fields.title !== 'string') {
  throw new Error("The 'fields' object must contain a 'title' key with a string value.");
}
```